src/Credentials/CredentialProvider.php

```diff
@@ -267,6 +265,27 @@ public static function ecsCredentials(array $config = [])
     }
 
     /**
+     * Remote credentials provider returns a credentials provider for the
+     * default remote endpoints such as EC2 or ECS Roles.
+     *
+     * @param array $config Array of configuration data.
+     *
+     * @return InstanceProfileProvider|EcsCredentialProvider
+     * @see Aws\Credentials\InstanceProfileProvider for $config details.
+     * @see Aws\Credentials\EcsCredentialProvider for $config details.
+     */
+    public static function remoteCredentialsProvider(array $config = [])
+    {
+        $ecsCredentialsUri = getenv(EcsCredentialProvider::ENV_URI);
+
+        if (!$ecsCredentialsUri) {
+            return self::instanceProfile($config);
+        }
+
+        return self::ecsCredentials($config);
```

Comment:
That's true, however, for a default credential provider, it's aimed at a general credential chain that makes most sense instead of catching all possible scenarios. Especially, this a unified behavior across all SDKs for the situation that failing fetching ECS credentials will be followed with an Ec2 metadata credential attempt. Rather than return `ecsCredentials` solo, I'd prefer something like below:

``` php
return self::chain(
    self::ecsCredentials($config),
    self::instanceProfile($config)
);
```